### PR TITLE
refactor: decimal loss is paid by the user

### PR DESCRIPTION
**Pull request by nowooj**
## Summary of changes

<Describe summary of major changes here>

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [ ] Wrote tests
- [ ] Added a relevant changelog entry to CHANGELOG.md

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: contracts/dezswap_pair/src/contract.rs [hunk 1]
```diff
@@ -625,12 +627,16 @@ fn compute_offer_amount(
 
     let one_minus_commission = Decimal256::one() - commission_rate;
     let inv_one_minus_commission = Decimal256::one() / one_minus_commission;
+    let before_commission_deduction: Uint256 = ask_amount * inv_one_minus_commission;
```
Comment (jhlee-young) [inline contracts/dezswap_pair/src/contract.rs]: I guess this line also has decimal loss which needs ceiling..?

Comment (nowooj) [reply contracts/dezswap_pair/src/contract.rs]: https://github.com/dezswap/dezswap-contracts/pull/5/commits/ef6220819767f89067aaa82175659d61a9dd0d12

**Review by jhlee-young — APPROVED** [review]
LGTM

File: contracts/dezswap_pair/src/contract.rs [hunk 2]
```diff
@@ -640,7 +650,10 @@ fn compute_offer_amount(
         Uint256::zero()
     };
 
-    let commission_amount = before_commission_deduction * commission_rate;
+    let mut commission_amount = before_commission_deduction * commission_rate;
```
Comment (jhlee-young) [inline contracts/dezswap_pair/src/contract.rs]: Nit: `before_commission_deduction - ask_amount` also works

Comment (nowooj) [reply contracts/dezswap_pair/src/contract.rs]: https://github.com/dezswap/dezswap-contracts/pull/5/commits/fd5574d65542fcc9c2b3360ac92e7f8bc88c81d4

**Review by psy2848048 — APPROVED** [review]

**Pull request merged**